### ISSUE TITLE
ADVI should report estimates for deterministics and hide transformed RVs

### ISSUE THREAD

**Issue by fonnesbeck** [pymc-devs/pymc#1172]
ADVI presents auto-transformed estimates to users but does not report determinstics (including bounded  random variables on their original scale). This will be confusing to users (see #1170), and the determinsitic quantities are often of primary interest.

Once the refactoring of transformed variable names is complete (#1168) it may make this easier to do.

**Comment by taku-y** [comment]
I agree. I will work on this after refactoring of #1168.

**Comment by taku-y** [comment]
Solved with #1229. Closes.

**Comment by fonnesbeck** [comment]
I'm going to re-open this because it really ought to go beyond reporting just in `sample_vp`. As a rule, all variables specified by the user should be reported on their original scale, and automatic transformations should be hidden unless explicitly asked for. So, `means` and `sds` returned by `advi` ought to return the original variables and deterministics, as they are usually of interest to the user. Sorry for not being clear in the original issue.

**Comment by taku-y** [comment]
I understand what you said. I think it is difficult to get analytic solution of means and sds of inverse transformed Gaussian RVs. Instead, we can integrate `sample_vp()` into `advi()` and uses like this:

```
# Simple usage
trace = advi(draws=1000, *kwargs)

# Get also means and sds as in the current version of advi()
trace, advifit = advi(draws=1000, advifit=True, *kwargs)
```

What do you think of this idea?

**Comment by fonnesbeck** [comment]
Something like that might work. Another approach might be to make `ADVIfit` a richer data structure, which is something I am tinkering with now.

**Comment by taku-y** [comment]
We can easily add `trace` to `ADVIfit`, but I can't come up with other information to be added to `ADVIfit`. I think `trace` is what the user wants to know.

**Comment by twiecki** [comment]
Separately, I also thought it might be good to add ADVI for NUTS initialization, something like `NUTS(init='ADVI')`.

**Comment by taku-y** [comment]
Good idea. I will start to implement in a few days. Could you tell me a specific example where `NUTS(init='ADVI')` should work? It's for test.

**Comment by twiecki** [comment]
Not sure `init='ADVI'` is the best API yet as we also need to pass how many iters to run. Could do maybe just `advi_steps=0` (default). Here's a good example http://pymc-devs.github.io/pymc3/notebooks/stochastic_volatility.html#Fit-Model should be become a single call to `NUTS`.

**Comment by taku-y** [comment]
We could also give ADVI parameters as a `dict`. And thanks for pointing to an example.

**Comment by twiecki** [comment]
@taku-y I'm actually having doubts about my proposal as it's a divergence from the more functional API philosophy we have followed this far. For example, `find_MAP()` is always an extra call, not something we do inside of a sampler to initialize. Perhaps that means we should keep the same spirit with ADVI initialization. I do think the API could be improved, though. For example, if we stored `model.dict_to_array(sds)` in the `advi` object returned by `fit_advi()`, e.g. as `advi.cov` the syntax would look much cleaner.
